pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -112,6 +120,19 @@ type domainContext struct {
 	// cli options
 	versionPtr    *bool
 	hypervisorPtr *string
+	// CPUs management
+	// Array of the information about the resources used by Applications.
+	// Each element corresponds to an Application.
+	resourcesUsedByVMs map[uuid.UUID]ResourcesUsedByVM
+	// Locks access to the resourcesUsedByVMs array
+	vmResourcesLock sync.Mutex
+	// Number of CPUs reserved for the EVE services and the Applications with no pinned CPUs
+	cpusReservedForEve int
+	// Map of the CPUs used for pinning.
+	// Indexed by CPU number. Stores `true` if the CPU is pinned to an Application, `false` otherwise.
+	hostCpusPinned map[int]bool
```

Comment:
Does hostCpusPinned have concurrent access from different goroutines?

Reply:
Yes, it has. And access to it is locked by the same lock now: `vmResourcesLock`

Reply:
This should also be left open until it has been resolved.

Reply:
What kind of resolution are we expecting here? A dedicated data structure for this field (along with the `resourcesUsedByVMs` field)?

Reply:
In addition to the locking part, I see several comments which refer to this as the shared set.
It might make sense to name it as such - since this is the set which will be shared by the app instances which do not have pinned CPUs. ("host pinned" doesn't indicate shared)

Reply:
This set is used by all the app instances: to check the state of the CPUs on the host (whether it's used exclusively by any VM or not).
Do you have an example of a good name for the shared set?

Reply:
sharedCpuSet? But that makes more sense if the resourcesUsedByVMs is named sometjing like pinnedCpuSet.

Reply:
The logic is moved into CPUAllocator. And it does not contain the set anymore.